deps/spidershim/src/v8trycatch.cc

```diff
@@ -41,7 +41,7 @@ struct TryCatch::Impl {
     // TODO: Propagate exceptions to the caller
     //       https://github.com/mozilla/spidernode/issues/51
   }
-  Isolate* Isolate() const { return isolate_; }
+  Isolate* Iso() const { return isolate_; }
```

Comment:
Do you mind pasting the test case here?  This program builds fine with gcc5 for me:

```cpp
class Foo;
class Bar {
  class Foo* Foo();
};
```